src/components/steps/step-sheet/DetailsTab.tsx

```diff
@@ -26,6 +27,15 @@ export function StepDetailsTab({ stepId, runId }: Props) {
 	const { data, isError, isPending, error } = useStepDetail({ stepId });
 	const { data: pipelineRunData } = usePipelineRun({ runId });
 
+	const repository = pipelineRunData?.body?.code_reference?.body?.code_repository;
+
+	const { data: codeRepositoryData } = useCodeRepositoryList(
```

Comment:
Instead of using the list endpoint, we can use the `/repository/{id}` endpoint for it cant we? It feels more suitable, because we're not really interested in the list, and it's hardcoded to index 0 

`const repositoryMetadata = codeRepositoryData?.items?.[0]?.metadata?.config;`